torch/csrc/jit/pybind_utils.h

```diff
@@ -37,14 +39,20 @@ namespace jit {
 // locations in libtorch code rather than user code.
 
 using tracer::TypedStack;
-struct TypedIValue : public std::pair<IValue, TypePtr> {
+struct TypedIValue : public std::pair<IValue, MatchTypeReturn> {
   using pair::pair;
 
+  TypedIValue(IValue value, TypePtr type)
+      : std::pair<IValue, MatchTypeReturn>(value, MatchTypeReturn(type)) {}
+
   IValue& ivalue() {
     return this->first;
   }
   TypePtr& type() {
-    return this->second;
+    return *this->second.type;
+  }
+  std::string error() {
+    return this->second.errMsg;
```

Comment:
It's confusing that TypedIValue is potentially not typed, and has an error message. Especially since typed ivalue's are nested, so you might have a tuple that contains another element that has an error message somewhere. Can you think of another approach?